### PR TITLE
[#467] Setup objecttypes through django-setup-configuration

### LINKED ISSUE #467: Add support for Objecttype-configuration via django-setup-configuration
### Thema / Theme

Objecten API

### Omschrijving / Description

Add django-setup-configuration to configure objecten to objecttypen

### Toegevoegde waarde / Added value

_No response_

### Aanvullende opmerkingen / Additional context

_No response_

### PR DESCRIPTION
**Pull request by SonnyBA**
Fixes #467 

**Changes**

Allows the configuration of `Service`'s with django-setup-configuration to setup a connection with objecttype(s)

### CODE REVIEW

**Comment by SonnyBA** [comment]
I haven't prefixed https://github.com/maykinmedia/objects-api/pull/492/commits/e49589944e93c58c8b0cb8eea16b0815f95540fb (with the ticket number) as I will cherry pick those for https://github.com/maykinmedia/objects-api/pull/491

File: docker/setup_configuration/data.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+zgw_consumers_config_enable: true
+zgw_consumers:
+  services:
+  - identifier: objecttypes-api
+    label: Objecttypes API
+    api_root: http://objecttypes.local/api/v1/
+    api_connection_check_path: objecttypes
+    api_type: orc
+    auth_type: api_key
```
Comment (stevenbal) [inline docker/setup_configuration/data.yaml:9]: let's also set the headers to make it a bit more of a real example:

```
header_key: Authorization
header_value: Token b9f100590925b529664ed9d370f5f8da124b2c20
```

**Review by stevenbal — CHANGES_REQUESTED** [review]

File: src/objects/setup_configuration/models/objecttypes.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+from django_setup_configuration.fields import DjangoModelRef
```
Comment (stevenbal) [inline src/objects/setup_configuration/models/objecttypes.py:1]: Both `models/` and `steps/` are missing `__init__.py`, not sure if it causes errors, but I do see warnings about it: 

```
WARNINGS:
?: (utils.W001) Directory %s does not contain an `__init__.py` file
        HINT: Consider adding this module to make sure tests are picked up
?: (utils.W001) Directory %s does not contain an `__init__.py` file
        HINT: Consider adding this module to make sure tests are picked up
```

File: src/objects/core/models.py [hunk 3]
```diff
@@ -41,8 +42,14 @@ def url(self):
         # zds_client.get_operation_url() can be used here but it increases HTTP overhead
         return f"{self.service.api_root}objecttypes/{self.uuid}"
 
-    def clean(self):
+    def clean_fields(self, exclude: Optional[Iterable[str]] = None) -> None:
```
Comment (stevenbal) [inline src/objects/core/models.py:45]: Minor, but we could use this syntax:

```suggestion
    def clean_fields(self, exclude: Iterable[str] | None = None) -> None:
```

File: src/objects/core/tests/files/objecttypes_idempotent.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+objecttypes_config_enable: true
```
Comment (stevenbal) [inline src/objects/core/tests/files/objecttypes_idempotent.yaml:1]: Do we need separate files for these tests? Most of them seem the same

**Review by stevenbal — APPROVED** [review]
Minor comments

File: src/objects/core/tests/files/objecttypes_idempotent.yaml (hunk 4 above)
Comment (SonnyBA) [reply src/objects/core/tests/files/objecttypes_idempotent.yaml:1]: Removed the idempotent test file, the other files are quite different from each other

**Pull request merged**